Exception in require-from-string on Webpack: `Can't resolve 'module'`

Using solc installed via npm or by including a js script with nuxtjs

```
import * as wrapper from 'solc/wrapper';
const solc = wrapper(window.Module);
```
 throws the following error
```
Module not found: Error: Can't resolve 'module' in 'C:\Code\NFTCalendar\NextFlip\node_modules\require-from-string'
```
apparently, this has been reported on the require-from-string repo 
https://github.com/floatdrop/require-from-string/issues/18

downgrading to require-from-string@1.2.1 does seem to fix the issue

but then it will throw
```
TypeError: Cannot use 'in' operator to search for '_solidity_version' in undefined
    at setupMethods (wrapper.js?92a3:10)
    at eval (index.vue?6ced:14)
    at Module../node_modules/babel-loader/lib/index.js?!./node_modules/vue-loader/lib/index.js?!./pages/index.vue?vue&type=script&lang=js& (index.js:11)
    at __webpack_require__ (runtime.js:854)
    at fn (runtime.js:151)
    at eval (index.vue?ff57:1)
    at Module../pages/index.vue?vue&type=script&lang=js& (index.js:47)
    at __webpack_require__ (runtime.js:854)
    at fn (runtime.js:151)
    at eval (index.vue?b7d3:1)
    ```

I'm having the same problem as you and I haven't solved it, I think it's probably because I'm not familiar with the webpack settings, because many people don't report this error, my "require-from-string":"^2.0.2" is the latest version.

```
index.js:3 Uncaught Error: Cannot find module 'module'
    at webpackMissingModule (index.js:3)
    at Object../node_modules/require-from-string/index.js (index.js:3)
    at __webpack_require__ (bootstrap:24)
    at fn (hot module replacement:61)
    at Object../node_modules/solc/wrapper.js (wrapper.js:3)
    at __webpack_require__ (bootstrap:24)
    at fn (hot module replacement:61)
    at Module../src/page/deploy/index.js (index.js:18)
    at __webpack_require__ (bootstrap:24)
    at fn (hot module replacement:61)
webpackMissingModule @ index.js:3
./node_modules/require-from-string/index.js @ index.js:3
__webpack_require__ @ bootstrap:24
fn @ hot module replacement:61
./node_modules/solc/wrapper.js @ wrapper.js:3
__webpack_require__ @ bootstrap:24
fn @ hot module replacement:61
./src/page/deploy/index.js @ index.js:18
__webpack_require__ @ bootstrap:24
fn @ hot module replacement:61
./src/index.js @ immutable.js:19
__webpack_require__ @ bootstrap:24
(anonymous) @ startup:6
(anonymous) @ startup:6
index.js:548 [webpack-dev-server] Hot Module Replacement enabled.
index.js:548 [webpack-dev-server] Live Reloading enabled.
index.js:558 [webpack-dev-server] Errors while compiling. Reload prevented.
logger @ index.js:558
(anonymous) @ index.js:728
error @ index.js:177
errors @ index.js:168
(anonymous) @ socket.js:46
client.onmessage @ WebSocketClient.js:35
index.js:558 [webpack-dev-server] ERROR in ./node_modules/require-from-string/index.js 3:13-30
Module not found: Error: Can't resolve 'module' in '/Users/developer/Documents/GitHub/golddog/study/node_modules/require-from-string'
```

--- a/package.json
+++ b/package.json
@@ -51,7 +51,6 @@
     "follow-redirects": "^1.12.1",
     "js-sha3": "0.8.0",
     "memorystream": "^0.3.1",
-    "require-from-string": "^2.0.0",
     "semver": "^5.5.0",
     "tmp": "0.0.33"
   },

--- a/wrapper.js
+++ b/wrapper.js
@@ -1,6 +1,6 @@
 const assert = require('assert');
 const translate = require('./translate.js');
-const requireFromString = require('require-from-string');
+const Module = require('module');
 const https = require('follow-redirects').https;
 const MemoryStream = require('memorystream');
 const semver = require('semver');
@@ -335,7 +335,16 @@ function setupMethods (soljson) {
         } else {
           response.pipe(mem);
           response.on('end', function () {
-            cb(null, setupMethods(requireFromString(mem.toString(), 'soljson-' + versionString + '.js')));
+            // Based on the require-from-string package.
+            const soljson = new Module();
+            soljson._compile(mem.toString(), 'soljson-' + versionString + '.js');
+            if (module.parent && module.parent.children) {
+              // Make sure the module is plugged into the hierarchy correctly to have parent
+              // properly garbage collected.
+              module.parent.children.splice(module.parent.children.indexOf(soljson), 1);
+            }
+
+            cb(null, setupMethods(soljson.exports));
           });
         }
       }).on('error', function (error) {

--- a/wrapper.js
+++ b/wrapper.js
@@ -1,6 +1,6 @@
 const assert = require('assert');
 const translate = require('./translate.js');
-const Module = require('module');
+const Module = module.constructor;
 const https = require('follow-redirects').https;
 const MemoryStream = require('memorystream');
 const semver = require('semver');

